Code/Framework/AzCore/AzCore/Jobs/JobGraph.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+// NOTE: If adding additional header/symbol dependencies, consider if such additions are better
+// suited in the private CompiledJobGraph implementation instead to keep this header lean.
+#include <AzCore/Jobs/Internal/JobTypeEraser.h>
+#include <AzCore/Jobs/JobDescriptor.h>
+#include <AzCore/std/containers/array.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/parallel/binary_semaphore.h>
+
+namespace AZ
+{
+    namespace Internal
+    {
+        class CompiledJobGraph;
+    }
+    class JobExecutor;
+    class JobGraph;
+
+    // A JobToken is returned each time a Job is added to the JobGraph. JobTokens are used to
+    // express dependencies between jobs within the graph.
+    class JobToken final
```

Comment:
Nitpick: It seems it's not needed to hold on the JobToken, but it would be good to clarify in the description that fire-and-forget jobs are supported and the token doesn't need to be stored.